src/Workspaces/Core/Portable/Workspace/Host/Status/ISolutionStatusService.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.CodeAnalysis.Host
+{
+    /// <summary>
+    /// Provides solution status
+    /// </summary>
+    internal interface ISolutionStatusService : IWorkspaceService
```

Comment:
>ISolutionStatusService [](start = 23, length = 22)

Agree with @CyrusNajmabadi here. These APIs are odd. Having service `IWorkspaceLoadStatusService` with methods:

```
Task<Solution> GetFullyLoadedSolutionAsync(CancellationToken cancellationToken);
Task<Project> GetFullyLoadedProjectAsync(ProjectId projectId, CancellationToken cancellationToken);
Task<bool> IsSolutionLoadedAsync(CancellationToken cancellationToken);
Task<bool> IsProjectLoadedAsync(ProjectId projectId, CancellationToken cancellationToken);
```

would make more sense to me.

Reply:
1. I would be ok with those methods.
2. I would also be ok with those methods just being exposed *directly* off of Workspace.  They dont' seem to need to be through a service IMO.  

Thoughts @tmat ?

Reply:
[2] I think that would be even better.

Reply:
Sounds good.  Thanks!